### ISSUE TITLE
we need a name

### ISSUE THREAD

**Issue by jedahan** [jedahan/living-room#1]
Any interesting terms in the comments that follows will be updated here:

- room
- space
- sees
- rumour
- whisper
- ask
- heard
- say
- message
- spread
- share
- symbol
- special
- turtle
- project
- recognize

things I'm not too keen on

- ~assert~
- ~retract~
- ~select~
- ~fact~
- ~squash~
- ~update~
- ~replace~
- ~rule~

**Comment by jedahan** [comment]
This is based off a fork of [roomdb](https://github.com/alexwarth/roomdb), exploring similar ideas of communicating about a space

**Comment by jedahan** [comment]
there will be things like cameras, connected to computers, helping fill the log with information like

> 'there are people in hopper'
> 'its very dark in the main room'
> 'the lovelace camera sees a whiteboard at (0.1, 0.9, 0.2)'
> 'there is a circle on a whiteboard at (20, 50)'
> 'the lovelace camera image is ', 0x16f666s777f77e55e....'

**Comment by jedahan** [comment]
Roomdb is based off datalog, which has ideas such as facts and rules. We've been using **assert**, **retract**, **select**, and **fact**, but I don't like them very much.

Fact implies truth, and I think the system should be able to break free from that.

It also implies implicitness and the ability to categorize. Sometimes categorization and naming can be violent. I get similar violent feelings around assert and retract.

Other words that are more in the direction - **rumour**, **whisper**, **ask**, **heard?**, **say**. **message** is okay too, though a bit generic. The project is super early so don't shy away from weird brainstorming, such as `spread(rumour)`, `quash(rumour)`, `heyDidyaHear?(rumour)`.

We might also need something for `update(rumour)` or `replace(rumour)`, though I feel like we can get by with a smaller surface area, and allow processes to do that work for us.

**Comment by jedahan** [comment]
Even though this issue is about vocabulary around the project in a hope of coming up with a good name, ideally the system would encourage non-text-based programming.

For examples, think of other symbols like 🐝 , and how the system could response to that.

I should be able to draw 🔴 ➡️ 😠  on a piece of paper, and the system draw angry faces on any red circles it sees. (of course, there has to be code and a process to recognize arrows, circles, etc).

Ideally we would be able to see and understand the code that are in these processes, within the system - if a projector is drawing an angry face on a red circle, we should be able to see the  🔴 ➡️ 😠 (physically, wherever it is written, either through a tiny video feed next to the face, or by arrows pointing us to walk over to it), and then the code that recognizes the arrow, etc, etc. 

There should be no hidden turtles. There should really be as little 'special' things as possible.

**Comment by kai-qu** [comment]
some terms I like: `convers(e/ation)`, `improv(isation)`, `act(ion)`, `react(ion)`, `dance`, `feel`, `say`, `speak`, `see`, `live`, `alive`, `move`, `motion`, `gesture`, `air`, `circle`, `synthesis`, `magic`, `spell`, `cast`, `mingle`, `twingle` (just kidding)

useful prefixes: `con-`, `inter-`, `tele-`, `syn-`, `sym-`, `trans-`, `peri-`

some mashups of the above: `Computair`, `Spacespeak`, `Living Room`

**Comment by modernserf** [comment]
Room + rumor = roomer.
A person working on roomer is a roomeur.

**Comment by sssoleileraaa** [comment]
Hey! This is better than I thought. Instead of naming just one thing, it looks like you're trying to come up with an entire vocabulary. 

So I was wondering if you could describe in the first comment how you plan to use these words. Like are they going to be variable and class names in code? Will they be commands that we send to camera systems, etc.? Do you foresee people shouting out commands to an AI, e.g. _turtle: whisper a rumor to a roomeur_. 

And what problem are you trying to solve? Are you hoping to reduce refactoring later on if you realize a certain word like 'turtle' is confusing people? Do you want to make sure you are using words that are more inclusive like instead of 'guy' you might use 'person'? 

It might be useful when suggesting a word to give an example of how it could be used, e.g.
_denizen: whisper to denizens of room called Lovelace, quote: your meeting is over, please exit the room promptly_.

**Comment by jedahan** [comment]
*tl;dr* immediate goal is naming the research project to differentiate it from other work, reduce preconceived notions, and to be less verbose than 'making the recurse center space programmable'

---

> denizen: whisper to denizens of room called Lovelace, quote: your meeting is over, please exit the room promptly.

Wow I love that last example @allieoop , and although this specific issue is mostly concerned with a project name, you hit the nail on the head that I am trying to come up with an entire vocabulary. To take a step back, the most immediate problem to solve is coming up with a project name that:

 - is less verbose than 'making the recurse center space programmable'
 - differentiates from other projects (like roomdb, spacebrew, realtalk, etc)
 - solidifies a vocabulary for core concepts
 - inspires collaboration
 - carries as little preconcieved notions as possible (in the spirit of hackerschool->recurse center) 

So that's the *why* for this issue, as of the *how*, there are a couple of things that can be done in parallel:
* Gather lots of words people like
* Try stringing words together and see how the concepts feel / fit
* Fill out words with examples (like you did) is the best idea in this thread so far - i'm definitely going to fill out more examples of how things would be used.
* Choose nonsense words (like early urbit.org did) as placeholders, instead of the current 
assert/retract/select/fact/room/db. Urbit literally made up new sounds like 'doz' 'nec' 'wod' 'bup' that were mapped to symbols like `%`, `!`, `~`, etc. Fun, but possibly distracting / far from what we need.

**Comment by davidbalbert** [comment]
@hypotext I think Living Room is a great name. Perfect pun :)

**Comment by sssoleileraaa** [comment]
Yes, +1 to Living Room and Computair

Because of the camera system, a creature with big eyes:
Aye-aye
Tarsier
Paraphromina
Cystisoma
Vampire Squid
Housefly
Geocoris
Deinopidae
Owl

What about arhouse? Like "our house". AR house.

Another +1 to Living Room

**Comment by rntz** [comment]
instead of `assert`, how about `learn`? and `retract` could be `unlearn` or `forget`. pretty straightforward.

**Comment by jedahan** [comment]
RealTalk uses concepts of `wishes` and `samples` as well

I like `learn`/`unlearn`/`share`, or `remember`/`forget`/`recall`...

learn and unlearn are a bit tricky for me to grok since client.learn() feels like the client is learning, though roomdb.learn makes some sense.

maybe client.share(``)...

**Comment by jedahan** [comment]
its living-room for now, also added https://github.com/jedahan/living-room-server, https://github.com/jedahan/living-room-js, and https://github.com/jedahan/living-room-database
